ballerina/types.bal

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/ai;
+import ballerina/sql;
+import ballerinax/postgresql;
+
+# Represents vector data without ID
+#
+# + id - Unique identifier
+# + embedding - Vector embedding array
+# + metadata - Optional metadata
+# + similarity - Optional similarity score
+public type SearchResult record {
+    string id;
+    string embedding;
+    string metadata?;

```

Comment:
Is this string?

Reply:
This record should not expose to the user. The embedding field is converted into float[] type in the implementation. Fixed in [4126203](https://github.com/ballerina-platform/module-ballerinax-ai.pgvector/pull/1/commits/4126203d03afd8cc655337b6b5f8197ec821c556)